src/GraphQL/schema/tags.graphql

```diff
@@ -0,0 +1,10 @@
+extend type Query {
+    tags: [Tag!]! @all(model: "Lab19\\Cart\\Models\\Tag")
+    tag(id: Int! @eq): Tag @find(model: "Lab19\\Cart\\Models\\Tag")
+}
+
+type Tag {
+    id: ID!
+    name: String
+    products: [Product!]! @hasMany(type: "paginator")
```

Comment:
@AlexeiaW this looks good, however the graphql api will ultimately need to support a method like the following, in order to keep 
```
productsByTag(tag: 1){
 
}
```

which effectively returns the paginated data as you already have it. 

I don't think there's much work to add this, you may just need to add a builder if you can't think of another cleaner way (much like this https://github.com/gernzy/gernzy-server/blob/master/src/GraphQL/schema/products.graphql#L7).

Reply:
Let's discuss this a bit further on Slack if you prefer.

Reply:
Ok, so I got the custom builder working, was quite a lot of trail and error to get the query to implement the correct sql. Just need your feedback on the current implementation